clients/src/main/java/org/apache/kafka/clients/SubscriptionPattern.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients;
+
+
+import com.google.re2j.Pattern;
+import com.google.re2j.PatternSyntaxException;
+
+public class SubscriptionPattern {
+    final private String pattern;
+
+    /**
+     * local variable to avoid multiple compile
+     */
+    private Pattern complitedPattern;
```

Comment:
Nit: typo

Reply:
Got it, thanks

Reply:
Remove the  RE2J and client part code, could you kindly help to review again? @kirktrue